### PR TITLE
add NaN edge blur protection

### PR DESCRIPTION
**Pull request by mholkesvik**
https://app.asana.com/0/795621786541567/848462337121489

### CODE REVIEW

File: samsts/smooth.py [hunk 1]
```diff
@@ -20,9 +20,20 @@ def smooth(filled_band, original_band, footprint_mask, dilation=0):
 
     Note - this helps reduce "speckle" in the final output
     """
-    gaussian = ndimage.filters.gaussian_filter(
-        filled_band, sigma=GAUSS_SIGMA, mode='reflect'
-    )
+
+    # some trickery to no NaN the edge pixels
+    # https://stackoverflow.com/a/36307291/10549650
+    nan_mask = filled_band != filled_band
```
Comment (max-nova) [inline samsts/smooth.py]: Er, why not just:

    nan_mask = np.isnan(filled_band)

File: samsts/smooth.py [hunk 2]
```diff
@@ -20,9 +20,20 @@ def smooth(filled_band, original_band, footprint_mask, dilation=0):
...
+
+    aux_v = filled_band.copy()
+    aux_v[nan_mask] = 0  # replace NaN with 0
+    aux_v = ndimage.gaussian_filter(aux_v, sigma=GAUSS_SIGMA, mode='reflect')
+
+    aux_w = 0 * filled_band.copy() + 1
```
Comment (max-nova) [inline samsts/smooth.py]: why not...

    aux_w = np.ones_like(filled_band)

File: samsts/process.py [hunk 3]
```diff
@@ -40,6 +40,7 @@ def process_full(landsat_fns, scenes_to_fill=None, out_dir=None):
     out_tifs = []
     for scene in scenes_to_fill:
         logging.info('Processing scene %s' % scene)
+        assert False, 'Stop'
```
Comment (mholkesvik) [inline samsts/process.py]:
```suggestion
```

File: samsts/run.py [hunk 4]
```diff
@@ -12,7 +14,7 @@
 )
 
 
-def run(job, path, row, scenes_to_fill, clear_cache=True):
+def run(job, path, row, scenes_to_fill, clear_cache=False):
```
Comment (mholkesvik) [inline samsts/run.py]: @max-nova switching to keep cache by default. same behavior as rest of our processes.

**Pull request merged**